### DISCUSSION TITLE
User defined settings/profiles

### DISCUSSION THREAD

**Discussion by marcin0klusek** [photopea/photopea#5923] [Ideas]
Hello there! 🙂

I use Photopea to create pixel art. Every time I run it, I must change the default modes of tools. 

What do you think about creating user profiles for tools? e.g. Default(created by devs), Pixel Art(created by user), Mockups(created by user). (free user = max. 3 profiles, premium user unlimited profiles?)

Also, it could store custom keyboard shortcuts(only for premium users?) 

It could be placed in the More tab?
![image](https://github.com/photopea/photopea/assets/34919312/36ba185c-1faf-4eab-b12c-f2572ac32b1c)

For pixel art, I set my tools to such mods:
 - B => Pencil Mode; By default its set to: Brush Tool
 - E => Eraser Tool with Mode: Pencil Tool
 - G => Paint Bucket Tool
 - Rest of tools: default settings
 - Brush: square, 1px(I set the square to help me visually):
![image](https://github.com/photopea/photopea/assets/34919312/3937e0af-3c78-43ec-95eb-84da7d47e3a4)

Regarding shortcuts, is it possible for you to add a shortcut for 'Duplicate layer' e.g. ALT+D? 
![image](https://github.com/photopea/photopea/assets/34919312/cfc2bf96-039b-4d49-8ac6-f9bde4764699)

**Comment by LazaroOnline** [comment]
The shortcut to duplicate a layer is `Ctrl/Cmd + J`, the same as in Photoshop.
The `J` is usually positioned far from your left hand, not being confortable to press,
but you can also press `ALT` while dragging a layer either from the canvas or from the layer's panel to duplicate it,
which is very confortable and used a lot.
This is also the same in Photoshop too.

**Comment by LazaroOnline** [comment]
Individual tools do have "Tool presets",
you can already create a preset for "1 pixel pencil" brush and earser.

Then to select a different "variant" of the same tool, 
like the "pencil" brush instead of the regular brush, 
just hold `Shift` while pressing the key shortcut for that tool.

To select the pencil brush it would be `Shift+B`
To select the bucket tool: `Shift+G`
Photopea will remember your latest tool selected until you change it again, 
so you don't need to use "Shift" constantly.

With his workflow I personally don't feel the need for profiles, 
and using these techniques gives me more flexibility.


<img width="262" alt="image" src="https://github.com/user-attachments/assets/82a40ead-b2ce-4910-ac16-c42e963539f0" />


The only thing is that it would be nice if the tool preset for the earser tool 
would also store the eraser's mode brush/pencil selected to load it latter,
which at this moment at least in photopea it does not (not sure about Photoshop).

<img width="276" alt="image" src="https://github.com/user-attachments/assets/189355e3-964b-476a-af36-9a963d15b84e" />

**Comment by photopea** [comment]
Photopea should remember the mode of the eraser tool now.
